packages/agents/src/functions/DevelopmentPlanner.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  Agent,
+  AgentFunctionResult,
+  LlmApi,
+  ChatLogs,
+  Tokenizer,
+  ChatMessageBuilder,
+  ChatMessage,
+} from "@evo-ninja/agent-utils";
+import { AgentFunctionBase } from "../AgentFunctionBase";
+import { AgentBaseContext } from "../AgentBase";
+
+interface FunctionParams {
+  goal: string;
+  context?: string;
+  summarizedInfo?: string;
+}
+
+export class DevelopmentPlanner extends AgentFunctionBase<FunctionParams> {
+  constructor(
+    private _llm: LlmApi,
+    private _tokenizer: Tokenizer
+  ) {
+    super();
+  }
+
+  get name() {
+    return "planDevelopmentTasks";
+  }
+
+  get description() {
+    return `Create execution steps to tackle any software development task`;
+  }
+
+  get parameters() {
+    return {
+      type: "object",
+      properties: {
+        goal: {
+          type: "string",
+        },
+        context: {
+          type: "string",
+        },
+        summarizedInfo: {
+          type: "string",
+        },
+      },
+      required: ["goal"],
+      additionalProperties: false,
+    };
+  }
+
+  buildExecutor(_: Agent<unknown>, context: AgentBaseContext) {
+    return async (
+      params: FunctionParams,
+      rawParams?: string
+    ): Promise<AgentFunctionResult> => {
+      const message = params.goal;
+
+      const content = `Given the goal: ${params.goal}. Please create a step-by-step plan to succesfully build what is asked.
+
+Your plan MUST consist in coding and testing iterations. Do not add anything into the plan that the user haven't asked for.
+
+### IMPORTANT
+Before starting to write any plan, make sure that you understand the requirements given in the goal;
```

Comment:
This is the first time we're mentioning "requirements", what is meant by this exactly?